### PR TITLE
feat: update firebase SDK to version 10.6.0

### LINKED ISSUE #27: fix: update Firebase SDK to the latest version
Update the FirebaseAnalytics SDK from the current version 10.3.0 to the latest version 10.7.0.

### PR DESCRIPTION
**Pull request by 1abhishekpandey**
# Description

- Updated debug messages by adding property information for delivered events.
- Added a debug log message in case an event name is not present and we are returning back.
- Refactored the `trackEvent` code by updating the way we handle different cases. The `trackEvent` method handles four sections:
    - Application Opened
    - E-Commerce Events
    - Custom Events
    - Sending of events to Firebase API
    - All four sections have been extracted into the respective methods for better readability.
- Updated the firebase SDK to the latest version 10.6.0
- Moved the sample events from AppDelegate to ViewController and also updated those sample events.
- There are no breaking changes.

### CODE REVIEW

**Review by pallabmaiti — APPROVED** [review]

File: Rudder-Firebase/Classes/RudderFirebaseIntegration.m [hunk 1]
```diff
@@ -58,70 +56,89 @@ - (void) processRudderEvent: (nonnull RSMessage *) message {
...
             [RSLogger logWarn:@"Message type is not recognized"];
         }
     }
 }
 
--(void) handleECommerce:(NSMutableDictionary *) params properties: (NSDictionary *) properties firebaseEvent:(NSString *) firebaseEvent {
+-(void) handleApplicationOpenedEvent {
+    NSString *firebaseEvent = kFIREventAppOpen;
+    NSMutableDictionary *params = [[NSMutableDictionary alloc] init];
+    [self makeFirebaseEvent: firebaseEvent params:params properties:nil];
```
Comment (desusai7) [inline Rudder-Firebase/Classes/RudderFirebaseIntegration.m]: Why are we sending empty params here ?

Why aren't we mapping properties to params ?

File: Example/Rudder-Firebase/RudderConfig/SampleRudderConfig.plist [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>WRITE_KEY</key>
+	<string></string>
+	<key>PROD_DATA_PLANE_URL</key>
+	<string></string>
```
Comment (desusai7) [inline Example/Rudder-Firebase/RudderConfig/SampleRudderConfig.plist]: Let's add the steps regarding how to use this file and also add some dummy values to make the example more meaningful

File: Example/Rudder-Firebase/_ViewController.m [hunk 3]
```diff
@@ -15,17 +15,244 @@ @interface _ViewController ()
...
+    [self checkoutStartedEvent];
+//    [self sendECommerceCustomAndScreenEvents];
+}
+
+-(void) identify {
+    NSMutableDictionary<NSString *,NSObject *> *traits = [[NSMutableDictionary alloc] init];
+    [traits setValue:@"random@example.com" forKey:@"email"];
+    [traits setValue:@"FirstName" forKey:@"fname"];
+    [traits setValue:@"LastName" forKey:@"lname"];
+    [traits setValue:@"1234567890" forKey:@"phone"];
```
Comment (desusai7) [inline Example/Rudder-Firebase/_ViewController.m:33]: why did we move them here ?

**Review by desusai7 — APPROVED** [review]

**Comment by 1abhishekpandey** [comment]
Closing this and going to create the release PR.

**Pull request closed without merging**